Tests/BotTests/GitHub/GitHubEventsTests.swift

```diff
@@ -2,83 +2,170 @@ import XCTest
+            },
+            assert: { events in
+                expect(events) == [Event.pullRequest(stubbedPullRequestEvent)]
+            }
         )
+    }
+
+    func test_handling_ping_event() {
+        perform(
+            when: { service, scheduler in
+
+                // NOTE: I'm unsure about the payload of this specific event
+                let req = request(
+                    with: .ping,
+                    signature: "9f64868b5cd91faa4c63589acff7286b16d289ae",
+                    payload: "{}"
+                )
 
-        let request = StubbedRequest(
-            headers: [GitHubService.HTTPHeader.event.rawValue: GitHubService.APIEvent.pullRequest.rawValue],
-            body: pullRequestEvent
+                let result = service.handleEvent(from: req).first()
+
+                scheduler.advance()
+
+                expect(result?.error).to(beNil())
+            },
+            assert: { events in
+                expect(events) == [Event.ping]
+            }
         )
+    }
 
-        var observedEvents: [Event] = []
+    func test_handling_unknown_event() {
+        perform(
+            when: { service, scheduler in
 
-        service.events
-            .observe(on: scheduler)
-            .observeValues { event in observedEvents.append(event) }
+                let req = request(
+                    withRawEvent: "anything_really",
+                    signature: "9f64868b5cd91faa4c63589acff7286b16d289ae",
+                    payload: "{}"
+                )
+
+                let result = service.handleEvent(from: req).first()
+
+                expect(result?.error) == .unknown
+            },
+            assert: { events in
+                expect(events) == []
+            }
+        )
+    }
+
+    func test_handling_untrustworthy_payload() {
+        perform(
+            when: { service, scheduler in
 
-        let result = service.handleEvent(from: request).first()
+                let req = request(
+                    withRawEvent: "anything_really",
+                    signature: "00faadd5dc1395b616a546ac6f67796409b4f839",
+                    payload: "{}"
+                )
 
-        scheduler.advance()
+                let result = service.handleEvent(from: req).first()
 
-        expect(result?.error).to(beNil())
-        expect(observedEvents) == [Event.pullRequest(pullRequestEvent)]
+                expect(result?.error) == .untrustworthy
+            },
+            assert: { events in
+                expect(events) == []
+            }
+        )
     }
+}
 
-    func test_handling_ping_event() {
-        let service = GitHubService()
-        let scheduler = TestScheduler()
+extension GitHubEventsTests {
+    
+    static let signatureToken = "Wall-E"
+
+    private func signature(for data: Data) -> String {
+        return try! HMAC(key: GitHubEventsTests.signatureToken, variant: .sha1)
+            .authenticate(data.bytes)
+            .toHexString()
+    }
```

Comment:
Is this private method still used now that we use hardcoded signatures in fixtures?

Reply:
It's useful if we need to update/add new signatures that's why I kept it, I can leave a comment.

Reply:
I see. Well to be fair we could just do like we do when re recording snapshot tests, but manually: run them, see them fail, and copy/paste the "actual" value obtained during test run in place of the "expected" value… no need for the private method for that 😅

Reply:
👍 I will remove it then.